src/gfn/states.py

```diff
@@ -989,220 +934,40 @@ def extend(self, other: GraphStates):
         Args:
             other: GraphStates object to concatenate with.
         """
-        if len(self) == 0:
-            # If self is empty, just copy other
-            self.tensor = self._clone_batch(other.tensor)
-            self.batch_ptrs = other.batch_ptrs.clone()
-            if other._log_rewards is not None:
-                self._log_rewards = other._log_rewards.clone()
-            return
-
-        self_x, other_x = self.tensor.x, other.tensor.x
-        self_edge_index, other_edge_index = (
-            self.tensor.edge_index,
-            other.tensor.edge_index,
-        )
-        self_edge_attr, other_edge_attr = self.tensor.edge_attr, other.tensor.edge_attr
```

Comment:
for reference I am talking about this logic here

I think it makes sense for us to retain these methods, no? @younik

Reply:
Where do you want to keep these methods? We are not using GeometricBatch anymore for stacking/extending, but just relying on numpy objects now